### PR TITLE
apmpackage: copy full field hierarchy

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Fix integration package index template mappings by copying the full field hierarchy to the integration package, instead of just flattened leaf fields. Copying only leaf fields means we miss inherited group properties such as `dynamic`.

- set a default `dynamic` property for each data stream mapping
- add `ecs.version` to base fields

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~

## How to test these changes

1. Install the integration
2. Run apm-server with data streams (e.g. `-E apm-server.data_streams.enabled=true`)
3. Send some HTTP transactions with request headers, check that APM Server indexes them successfully
4. Check that the request headers are not indexed

## Related issues

https://github.com/elastic/apm-server/issues/5707

### CODE REVIEW

File: apmpackage/apm/data_stream/internal_metrics/manifest.yml [hunk 1]
```diff
@@ -2,3 +2,9 @@ title: APM internal metrics
 type: metrics
 dataset: apm.internal
 ilm_policy: metrics-apm.internal_metrics-default_policy
+elasticsearch:
+  index_template:
+    mappings:
+      # Internal metrics should have all fields strictly mapped;
+      # we are in full control of the field names.
+      dynamic: strict
```
Comment (simitt) [inline apmpackage/apm/data_stream/internal_metrics/manifest.yml:10]: That would potentially lead to errors if the APM Server version > package version containing the templates. I'd be careful with using `strict` as long as the automatic package upgrade is not in place.

File: apmpackage/apm/data_stream/sampled_traces/manifest.yml [hunk 2]
```diff
@@ -9,3 +9,7 @@ elasticsearch:
       # global checkpoints as a way of limiting search
       # results.
       number_of_shards: 1
+    mappings:
+      # Sampled traces should have all fields strictly mapped;
+      # we are in full control of the field names.
+      dynamic: strict
```
Comment (simitt) [inline apmpackage/apm/data_stream/sampled_traces/manifest.yml:15]: same comment as above

File: apmpackage/apm/data_stream/internal_metrics/manifest.yml (hunk 1 above)
Comment (axw) [reply apmpackage/apm/data_stream/internal_metrics/manifest.yml:10]: Discussed offline, we'll plan to require Kibana >= APM Server, and rely on auto-upgrade: https://github.com/elastic/apm-server/issues/5948. If that doesn't pan out then we can revisit this.

**Review by simitt — APPROVED** [review]

**Comment by simitt** [comment]
Tested with BC3 - works as expected with package `0.4.0`

**Pull request merged**